server_status.py

```diff
@@ -0,0 +1,166 @@
+  line = inspect.currentframe().f_back.f_lineno
+  print time.strftime('%X') + ' - line ' + str(line + 1)
+
+def reset():
+  red.write(0)
+  green.write(0)
+  blue.write(0)
+
+# flash list of LEDs for individual periods and total duration (cycles rounded to multiple of len(leds) )
+def flash_colours(leds, duration, period):
+  n = len(leds)
+  for i in range(int( n * math.floor(.5 + duration/(n*n*period)) )):
+    leds[i%n].write(1)
+    time.sleep(period)
+    leds[i%n].write(0)
+    if n == 1:
+      time.sleep(period)
+
+def openmrs_internal_status(): # runs for ~1 second regardless of state
+  global openmrs_int
+  with open('/home/root/debian/home/buendia/server_status.txt', "r") as status_file:
+    openmrs_int_new = int(status_file.read())
+  if not openmrs_int_new == openmrs_int:  # state has changed
+    print time.strftime('%X') + ' - openmrs_int change: ' + str(openmrs_int) + ' > ' + str(openmrs_int_new)
+    reset()
+    openmrs_int = openmrs_int_new
+  # TASK LED's according to OpenMRS server status
+  if openmrs_int == 0:       # down - flash RED
+    flash_colours([red], 1, .5)
+  elif openmrs_int == 1:     # normal use - BLUE on
+    blue.write(1)
+    time.sleep(1)
+  elif openmrs_int == 2:     # activity  - blink BLUE on/off
+    flash_colours([blue], 1, .05)
+  elif openmrs_int == 3:     # backing-up - blink BLUE/GREEN
+    flash_colours([blue, green], 1, .1)
+  elif openmrs_int == 4:     # back-up failed - blink BLUE/RED
+    flash_colours([blue, red], 1, .5)
+  else:
+    time.sleep(1)
+
+# classify server repsonses - i.e. interpret top line of 'curl -Is <url>'
+def check_url(url):
+  global last_check
+  response = subprocess.check_output("curl --max-time 1 -Is " + url + " | head -n 1", shell=True)
+  if url == openmrs_url:
+    report()
+    last_check = time.time()
+  if response == '':               # no response
+    return [0, response]
+  elif response.find('200') > -1:  # positive response
+    return [1, response]
+  else:                            # other response (e.g. redirected) [Might have to debug this..]
+    return [2, response]
+
+
+# Test LEDs, then RED on
+report()
+flash_colours([red, green, blue], 3, .05)
+time.sleep(1)
+red.write(1)
+
+
+## STATUS CODES
+# 0 - down
+# 1 - running
+# 2 - activity
+# 3 - backing-up
+# 4 - backup failed
+
+
+# MAIN LOOP
+
+## LOGIC SUMMARY
+# has Buendia been detected yet?
+#   no - is it up now?
+#     no - sleep for 5s
+#     yes - change state to detected
+#   yes - has OpenMRS been detected yet?
+#     no - is it up now?
+#       no - flash RED/BLUE for 5s
+#       yes - change state to detected & BLUE on
+#     yes - has minute passed since last external check?
+#       no - check OpenMRS internal status & report accordingly
+#       yes - has OpenMRS stopped responding to ping?
+#         yes - flash RED for 60s till next ping check
+
+
+print '*** START MAIN LOOP ***'
+
+while True:
+  if time.time() - last_check > 10800:  # truncate log file every 3 hours
```

Comment:
Unless I'm misunderstanding, last_check is reset every time the server is checked, so this condition will be true 3 hours after the server was started up and will continue to be true.

So once 3 hours has passed, the log file will be truncated with every iteration of the loop.

If we really wanted to truncate it every 3 hours, I think this needs another variable to track when the log file was last truncated.

Though it's probably fine to keep it truncated to the last 1000 lines, rather than truncating it every 3 hours.